src/server/socket.ts

```diff
@@ -1,16 +1,75 @@
 import * as socketio from "socket.io";
 import { getToken } from "next-auth/jwt";
 import { v4 as uuidv4 } from "uuid";
-import { parse } from "cookie";
+import { parse, serialize } from "cookie";
 // Due to the way Next creates a production application, we need to import knex
 // via a relative path, not using "@"
 import { knex } from "../knex/knex";
 import Room from "../models/Room";
+import GraspReaction from "../models/GraspReaction";
 
 // Develop from route description using /rooms/:rid/:admin?
 // http://forbeslindesay.github.io/express-route-tester/
 const roomRegEx = /^\/rooms\/(?:([^\/]+?))(?:\/([^\/]+?))?\/?$/i;
 
+interface GraspReactionWithCount extends GraspReaction {
+  count: number;
+}
+
+async function getActiveGraspReactionCount(
+  roomId: number,
+  timeInMinutes: number = 5,
+) {
+  const currentDate = new Date(Date.now()); // Use Date.now() for testing
+
+  // Grab the latest reaction from a specific user
+  const latestReactions = await GraspReaction.query()
+    .select(knex.raw("max(id) as id")) // Assumption: Reactions from oldest to newest
+    .whereNotNull("anon_grasp_user_id")
+    .groupBy("anon_grasp_user_id");
+
+  // Filter by is_active, room, and user id
+  const activeReactions = await GraspReaction.query()
+    .where("is_active", true)
+    .andWhere("room_id", roomId)
+    .whereIn(
+      "id",
+      latestReactions.map((reaction) => reaction.id),
+    ); // only latest reactions from unique users
+
+  // Calculate the count based on activeReactions (linear decay)
+  const levelCount: GraspReactionWithCount[] = (await GraspReaction.query()
```

Comment:
Yeah, I don't think calculating this in the database would be very beneficial since the relevant entries were already found and finding and summing them in one operation might lead to very complex queries. I've modified it so that it's done in JS and looks simpler.